src/commands/up.ts

```diff
@@ -70,15 +94,23 @@ async function up_action(options: UpOptions, ...components: string[]): Promise<v
     });
   }
 
-  let targetGraph = await environment.getGraph(options.environment, command_helper.componentStore, options.debug);
+  const envName = options.environment || uniqueNamesGenerator({
+    dictionaries: [animals],
+    length: 1,
+    separator: '-',
+    style: 'lowerCase',
+    seed: Deno.cwd(),
```

Comment:
This also fixes this. https://github.com/architect-team/arcctl/issues/145

Reply:
Actually I think this instead should be `components.join(',')`. If this command is aliased it could be run from any location. The cwd is not a good representation of what the user is trying to do. However the components are.

Reply:
I thought about that option, but `components` is likely going to be an array of relative paths which means their values change if you run from a different directory too. I decided it was just easier to use Deno.cwd() given that.

Reply:
Their values should not change if run from a different directory though. That is only true if you use relative paths to the current directory which generally speaking is not common. Even all architect customers generally use absolute or `~` in my experience. So if you run the same command N times you get the same env name. For instance this is the alias we generally use for dev on the other cli.  `alias architect="XDG_CONFIG_HOME=~/.config/architect/local ~/Architect/code/architect-cli/bin/dev"` This is the one I use for `arcctl`. `alias arcctl="deno run -A /home/muesch/Architect/code/arcctl/main.ts"` In both cases no matter where I run that the env name would be the same. 

This would even be better for teams like Twenty who use makefiles since the makefiles all use the same component with the relative path, but they actually have multiple directories they can run make from. The makefiles just chain from one to the other. So the `cwd` changes but the commands don't.

Reply:
I feel like you may be confusing cwd with the binary path? Those aliases and the location of the binary don’t have anything to do with the cwd. That’s the directory I’m currently cd’d into in my terminal.

Reply:
Some examples
Alias
```
echo "export my_component_run=arcctl up ~/code/my_component ~/code/my_component2
cd ~
my_component_run
// cwd is ~
// components are  "~/code/my_component,~/code/my_component2"
cd ~/code/my_component
my_component_run
// cwd is ~/code/my_component
// components are  "~/code/my_component,~/code/my_component2"
```
The component path stays static but the `cwd` is dynamic. The same holds true for `package.json`
```
{
  "scripts": {
    "dev": "arcctl up ."
  }
}
```
In this case both the `cwd` and `components` will be the same since they are executed from the same directory. So either work well.

Makefiles are also interesting for monorepos.
Root Makefile
```
%:
	$(MAKE) -C dev $*
```
This essentially sits in the root folder and proxies all commands to a sub directory. This is not an uncommon pattern.
If we look at the subdirectory Makefile it would look like
```
up:
	@arcctl up ../server ../client
```
If we use those files
```
cd ~/project_root
make up
// cwd ~/project_root
// components ../server,../client
cd ~/project_root/dev
make up
// cwd ~/project_root/dev
// components ../server,../client
```

The point is that the `cwd` is a very dynamic piece of information, while the components are not. The only time the `cwd` is less dynamic than the component paths is when the components themselves change. Which in my experience is not all that common relative to running commands in different places.

Reply:
That's an odd use-case. Do you use aliases to setup our architect environment like that? I've never done anything other than `arcctl up ./dir`. I also often interchange between specifying the path to the architect.yml and just the directory which would lead to different values for "components". 

It also seems to me that if we're looking for determinism then we should allow flags to force that determinism rather than bend over backwards to make the auto-generated determinism be perfect. Should we allow you to specify an environment name that doesn't exist with the `-e` flag to help with this use case?

Reply:
Yeah its a very common dev flow. I have seen a lot of projects do stuff like this. The idea is to have a simple thing people can remember to start their dev env. Using something like the package.json file also means that you can modify the flow without developers changing their habits.

The `-e` flag creating the env would also be good. Just also making the argument that the general point of my argument is not to be perfect but rather to cover what is in my experience the more common usecase by default.

Reply:
Alright, I updated the generation to match this behavior. Let me know if anything else needs to be changed to get this merged.